src/coreclr/src/pal/src/init/pal.cpp

```diff
@@ -1376,9 +1376,12 @@ bool GetEntrypointExecutableAbsolutePath(PathCharString& entrypointExecutable)
         }
         else
         {
+            PathCharString pathString;
+            pathString.Set(path, strlen(path));
+
             entrypointExecutable.Set(cwd, strlen(cwd));
             entrypointExecutable.Append('/');
-            entrypointExecutable.Append(path);
+            entrypointExecutable.Append(pathString);
```

Comment:
`Append` also has the variant with `char*, length` args. Could you please use that form instead of creating another instance of PathCharString?:
```suggestion
            entrypointExecutable.Append(path, strlen(path));
```

Reply:
I overlooked the `StackString<,>` definition; pushed the single line change without extra instance.